src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/allocation/AllocationRequest.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.deliusapi.dto.v1.allocation
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.deliusapi.validation.AllocationReasonCode
+import uk.gov.justice.digital.hmpps.deliusapi.validation.DeliusDate
+import uk.gov.justice.digital.hmpps.deliusapi.validation.StaffCode
+import uk.gov.justice.digital.hmpps.deliusapi.validation.TeamCode
+import java.time.LocalDateTime
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.PastOrPresent
+
+data class AllocationRequest(
+
+  @Schema(title = "The staff code of the allocation")
+  @StaffCode
+  @field:NotBlank
+  val staffCode: String,
+
+  @Schema(title = "The team code of the allocation")
+  @TeamCode
+  @field:NotBlank
+  val teamCode: String,
+
+  @Schema(title = "The date time of the allocation")
+  @DeliusDate
+  @field:PastOrPresent
+  val datetime: LocalDateTime,
+
+  @Schema(title = "The reason for the allocation")
```

Comment:
Sorry I wasn't super clear on this comment, I just meant update the `@Schema` annotation - so anyone looking at Swagger know they have to pass a reference data code and not just any string  😄 
```suggestion
  @Schema(title = "The reason code for the allocation")
```